### PR TITLE
feat(xmllib): add examples to docstrings (DEV-4423)

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**

### CODE REVIEW

File: src/dsp_tools/xmllib/__init__.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 from .models.dsp_base_resources import VideoSegmentResource as VideoSegmentResource
 from .models.resource import Resource as Resource
 from .models.root import XMLRoot as XMLRoot
-from .value_checkers import find_geometry_problem as find_geometry_problem
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/xmllib/__init__.py:16]: Since they don't write the geometry by hand anymore, we can remove this function.

File: src/dsp_tools/xmllib/helpers.py [hunk 2]
```diff
@@ -28,6 +28,47 @@ def create_label_to_name_list_node_mapping(
...
+        ]
+        ```
+
+        ```python
+        result = xmllib.create_label_to_name_list_node_mapping(
+            project_json_path="project.json",
+            list_name="listName",
+            language_of_label="de",
+        )
+        # -> { "Knoten 1": "n1", "Knoten 2": "n2" }
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/xmllib/helpers.py:70]: I asked Noemi about the display of the result and she was happy with this.
I also considered the more verbose but also more explicit:
`# result = {}`
This would be fine by me too if you prefer it.

File: src/dsp_tools/xmllib/models/resource.py [hunk 3]
```diff
@@ -846,6 +1140,7 @@ def add_richtext(
         Conversions:
             By default, replace newline characters inside the text value with `<br/>`, which preserves the linebreak.
             Without this replacement, the newline would disappear, because `\\n` is meaningless in an XML file.
+            [See the function documentation for details](https://docs.dasch.swiss/latest/DSP-TOOLS/xmllib-api-reference/value-converters/#xmllib.value_converters.replace_newlines_with_tags)
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/models/resource.py:1143]: "the function documentation" is a bit vague.

```suggestion
            [Click here for more details](https://docs.dasch.swiss/latest/DSP-TOOLS/xmllib-api-reference/value-converters/#xmllib.value_converters.replace_newlines_with_tags)
```

File: src/dsp_tools/xmllib/models/resource.py [hunk 4]
```diff
@@ -858,6 +1153,22 @@ def add_richtext(
...
+                value="line 1\\nline 2",
+            )
+            ```
+
+            ```python
+            resource = resource.add_richtext(
+                prop_name=":propName",
+                value="line 1\\nline 2",
+                newline_replacement=xmllib.NewlineReplacement.PARAGRAPH,
+            )
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/models/resource.py:1170]:
```suggestion
            )
             # ->  <p>line 1</p><p>line 2</p>
```

File: src/dsp_tools/xmllib/models/resource.py [hunk 5]
```diff
@@ -858,6 +1153,22 @@ def add_richtext(
 
         Returns:
             The original resource, with the added value
+
+        Examples:
+            ```python
+            resource = resource.add_richtext(
+                prop_name=":propName",
+                value="line 1\\nline 2",
+            )
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/models/resource.py:1162]:
```suggestion
            )
           # ->  line 1<br/>line 2
```

File: src/dsp_tools/xmllib/models/resource.py [hunk 6]
```diff
@@ -884,6 +1195,7 @@ def add_richtext_multiple(
         Conversions:
             By default, replace newline characters inside the text value with `<br/>`, which preserves the linebreak.
             Without this replacement, the newline would disappear, because `\\n` is meaningless in an XML file.
+            [See the function documentation for details](https://docs.dasch.swiss/latest/DSP-TOOLS/xmllib-api-reference/value-converters/#xmllib.value_converters.replace_newlines_with_tags)
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/models/resource.py:1198]: same as above

File: src/dsp_tools/xmllib/models/resource.py [hunk 7]
```diff
@@ -924,6 +1244,7 @@ def add_richtext_optional(
...
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/models/resource.py:1247]: same as above

File: src/dsp_tools/xmllib/value_converters.py [hunk 8]
```diff
@@ -18,6 +18,24 @@ def convert_to_bool_string(value: Any) -> str:
...
+        # -> "true"
+        ```
+
+        ```python
+        result = xmllib.convert_to_bool_string("nein")
+        # -> "false"
+        ```
+
+        ```python
+        # because this is not an accepted value, it is returned as a string
```
Comment (jnussbaum) [inline src/dsp_tools/xmllib/value_converters.py:34]: Not part of this PR, but here it stuck me that this behaviour is contra-intuitive.
There mere fact that you had to write this explanation shows that it is not natural.

**Review by jnussbaum — APPROVED** [review]
